### DISCUSSION TITLE
changing Mesh color without changing material-color (support)

### DISCUSSION THREAD

**Discussion by jasan-s** [pmndrs/drei#674] [Q&A]
_NOTE: The discord invite link for support is invalid_

I have a model in which multiple mesh geometry share the same material.  If i add value to ` material-color={'#ffffff}` the  color changes for all mesh that share the material name. Is it possible to change the color of each mesh individually without effecting other meshes? 

I tried providing simple color prop to mesh but doesnt do anything. 

i.e 
```
    <mesh
        geometry={nodes.Cylinder.geometry}
        material={materials.metal}
        position={[0, 1.35309779644, 0.06925085187]}
        scale={[1, 0.771013319492, 1]}
      />
      <mesh
        geometry={nodes.Cylinder001.geometry}
        material={materials.metal}
        position={[0, 1.347179532051, 0.065558932722]}
        scale={[1, 0.771013319492, 1]}
      />
```

**Comment by joshuaellis** [comment]
They share the same material. So changing one material changes them all. 

If you want to avoid this you can use the `clone` method which will keep all the properties but they will be a separate object:

```js 
<mesh
        geometry={nodes.Cylinder.geometry}
        material={materials.metal.clone()}
        position={[0, 1.35309779644, 0.06925085187]}
        scale={[1, 0.771013319492, 1]}
        material-color={0x0000ff}
      />
      <mesh
        geometry={nodes.Cylinder001.geometry}
        material={materials.metal.clone()}
        position={[0, 1.347179532051, 0.065558932722]}
        scale={[1, 0.771013319492, 1]}
        material-color={0x00ff00}
      />
```

**NOTE**: _this is an example, and you probably don't want to clone the materials in the render method because they'll clone each react-render of the component, i'm just showing the example_

**Reply by jasan-s** [reply]
@joshuaellis thanks I tried that but with state changes the color resets to the original material color.  Would you say the best way to customize each item color is to duplicate the material in the 3d program and assign it to each mesh separately ?  I assumed doing that would make the resulting 3d model FileSize large(Not a 3D Expert).  Maybe there is a more efficient method to to reuse the same material and assign it individually o each mesh Before the 3D Model gets to Three-Fiber . Any other suggestions or guide would be helpful
